lib/compat/wordpress-6.7/rest-api.php

```diff
@@ -114,3 +114,35 @@ function gutenberg_override_default_rest_server() {
 	return 'Gutenberg_REST_Server';
 }
 add_filter( 'wp_rest_server_class', 'gutenberg_override_default_rest_server', 1 );
+
+/**
+ * Add `object_subtype` to schema.
+ *
+ * @param array  $args Array of arguments for registering meta.
+ * @return array Modified arguments array including `object_subtype`.
+ */
+function gutenberg_add_object_subtype_to_schema( $args ) {
+	// Don't update schema when object_subtype isn't provided.
+	if ( ! isset( $args['object_subtype'] ) ) {
+		return $args;
+	}
+
+	$schema = array( 'object_subtype' => $args['object_subtype'] );
+	if ( ! is_array( $args['show_in_rest'] ) ) {
+		$args['show_in_rest'] = array(
+			'schema' => $schema,
+		);
+		return $args;
+	}
+
+	if ( ! empty( $args['show_in_rest']['schema'] ) ) {
+		$args['show_in_rest']['schema'] = array_merge( $args['show_in_rest']['schema'], $schema );
+	} else {
+		$args['show_in_rest']['schema'] = $schema;
+	}
+
+	return $args;
+}
+
+// Priority must be lower than 10 to ensure the object_subtype is not removed.
+add_filter( 'register_meta_args', 'gutenberg_add_object_subtype_to_schema', 5, 1 );
```

Comment:
@SantosGuillamot found that in Core, this value is unset so is not shown on the rest API.
https://github.com/WordPress/wordpress-develop/blob/30523ee4b6a0d33b1c2bd8c9ca4021b6256a96c3/src/wp-includes/meta.php#L1525

Are we making any breaking change with this filter?
ping @TimothyBJacobs

Reply:
It isn't a schema keyword, so it's a bid odd. Why are you needing to output this information at all though?

Reply:
Is needed to ensure in templates that could apply to any post type, we only load the fields without `object_subtype`.
If we register something like this:
```
register_meta(
		'post',
		'videogame_cover',
		array(
                        'object_subtype' => 'videogame',
			'show_in_rest'      => true,
			'single'            => true,
			'type'              => 'string',
			'default'           => 'https://images.igdb.com/igdb/image/upload/t_cover_big/co67qb.webp',
		)
	);
```
It would appear on Default Archive template or Single template.

We can land it without this check and get some feedback if this is an issue.

Reply:
Hm, I'm not sure I'm following completely. When you make your `OPTIONS` request to the specific post type endpoint, it takes into account the object subtype when fetching the list of meta fields.

Reply:
I've removed all that part. It seems that the `OPTIONS` request is filtering the correct meta fields as you mentioned.

In case is not enough, I'm happy to revisit in a follow-up PR.

Reply:
Nice. Sounds good.

Reply:
I've posted [this comment](https://github.com/WordPress/gutenberg/pull/64072#issuecomment-2353117590) trying to clarify the original use case.

Basically, we wanted a way to get the fields registered WITHOUT `object_subtype` to only show those in templates that can apply to any post type. Right now, the fields registered with `object_subtype => 'post'` are also shown.